src/main/java/io/lettuce/core/LMoveArgs.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2011-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lettuce.core;
+
+import io.lettuce.core.internal.LettuceAssert;
+import io.lettuce.core.protocol.CommandArgs;
+
+import static io.lettuce.core.protocol.CommandKeyword.*;
+
+/**
+ * Argument list builder for the Redis <a href="http://redis.io/commands/blmove">BLMOVE</a> and
+ * <a href="http://redis.io/commands/lmove">LMOVE</a> commands. Static import the methods from {@link Builder} and
+ * chain the method calls: {@code source(LEFT).destination(RIGHT)}.
+ *
+ * @author Mikhael Sokolov
+ */
+public class LMoveArgs implements CompositeArgument {
+
+    public enum Direction {
+        LEFT, RIGHT
+    }
+
+    private Direction source = Direction.RIGHT;
+
+    private Direction destination = Direction.LEFT;
+
+    /**
+     * Builder entry points for {@link LMoveArgs}.
+     */
+    public static class Builder {
+
+        /**
+         * Utility constructor.
+         */
+        private Builder() {
+        }
+
+        /**
+         * Creates new {@link LMoveArgs} setting {@literal SOURCE}.
+         *
+         * @return new {@link ZAddArgs} with {@literal SOURCE} set.
+         * @see LMoveArgs#source(Direction)
+         */
+        public static LMoveArgs source(Direction direction) {
```

Comment:
Let's name these methods `leftRight` and `rightLeft`. We don't need a method argument and we don't need to expose the `Direction` enum.

Reply:
I agree with the "all args constructor" approach.

Reply:
what about like this:
```
public static LMoveArgs move(Direction source, Direction dest) {
//...
}
```
The `source ` and `dest ` are required.

Reply:
My bad, the command documentation confused me to some extent. I need to reconsider my opinion here. Given that the command comes with some more dynamics, we indeed should keep `Direction` public. On the builder side, I'd still suggest to have `rightLeft` and `leftRight` entry methods plus @dengliming's proposal `LMoveArgs move(Direction source, Direction dest)`. 

Let's make the constructor of `LMoveArgs` `protected` so the class remains extensible.

Reply:
@dengliming @mp911de both proposals lgtm. But let's start with just static `leftLeft`, `leftRight`, `rightLeft` and `rightRight`. And yes, then just expose some existing or add new ones if needed.